legend-engine-xt-relationalStore-executionPlan/src/main/java/org/finos/legend/engine/plan/execution/stores/relational/RelationalExecutor.java

```diff
@@ -265,6 +266,15 @@ else if (var.getValue() instanceof PreparedTempTableResult && sqlQuery.contains(
             {
                 sqlQuery = sqlQuery.replace("(${" + var.getKey() + "})", ((PreparedTempTableResult) var.getValue()).getTempTableName());
             }
+            else if (var.getValue() instanceof RelationalResult && (sqlQuery.contains("inFilterClause_" + var.getKey() + "})") || sqlQuery.contains("${" + var.getKey() + "}")))
```

Comment:
This check looks more like a hack

Reply:
As discussed with @gs-gunjan and @davidharte-gs - we will approve/merge this for the next release, and will continue looking into how we can enhance the code to be more robust, and be able to remove this hack.